pandas/tests/indexes/test_base.py

```diff
@@ -237,61 +223,63 @@ def test_constructor_int_dtype_float(self, dtype):
         result = Index([0., 1., 2., 3.], dtype=dtype)
         tm.assert_index_equal(result, expected)
 
-    def test_constructor_int_dtype_nan(self):
+    @pytest.mark.parametrize("dtype,klass_or_raises", [
```

Comment:
Not sure how we feel about mixing types in `klass_or_raises` - could also be split into a boolean flag for raises and the klass, though the test would typically just use one or the other

Reply:
haha just made this comment